javatests/arcs/core/storage/database/DatabaseManagerTest.kt

```diff
@@ -0,0 +1,83 @@
+package arcs.core.storage.database
+
+import arcs.core.testutil.assertSuspendingThrows
+import arcs.jvm.storage.database.testutil.FakeDatabaseManager
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.mock
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@ExperimentalCoroutinesApi
+@RunWith(JUnit4::class)
+class DatabaseManagerTest {
+
+    private lateinit var mockDatabase: Database
+
+    private val databaseManager = FakeDatabaseManager()
+
+    private val dbNames = (1..100).map { "db$it" }
+
+    @Before
+    fun setup() {
+        mockDatabase = mock {}
+        databaseManager.registry.run {
+            dbNames.forEach {
+                register(it, true)
+            }
+        }
+    }
+
+    @Test
+    fun runOnAllDatabasesWaitsForAll() = runBlockingTest {
+        val visited = mutableListOf<String>()
+        databaseManager.runOnAllDatabases { name, _ ->
+            delay(1)
+            visited.add(name)
+        }
+        assertThat(visited).containsExactlyElementsIn(dbNames)
+    }
+
+    @Test
+    fun runOnAllDatabasesPropagatesException() = runBlockingTest {
+        val visited = mutableListOf<String>()
+
+        assertSuspendingThrows(FakeException::class) {
+            databaseManager.runOnAllDatabases { name, _ ->
+                if (name == "db42") {
+                    throw FakeException()
+                }
+                delay(1)
+                visited.add(name)
+            }
+        }
+        // Some number of additions may have proceeded.
+        assertThat(visited.size).isLessThan(42)
+    }
```

Comment:
I wonder if the correct behavior/end-result here would be to have `visited.size == 99`?  (ie. it runs on every database that it can run on.. and throws the exception later)  It might require some kind of aggregate exception, in case exceptions are thrown on multiple dbs.

Reply:
I was wondering that as well. I have a vague memory of having a conversation about this on the PR that introduced the change; I can't find it now though.

It seems like the right way to go, since we'll clear as much data as possible.

Reply:
It may have been that we decided to do the change later, since the current structure has only one database per manager anyway? Or something along those lines....

Reply:
I think I can make the change here; since this is just adding tests, I can change the nature of the PR to be "add this functional change + test"